File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/model/Match.kt
package dev.tpavani.octopoints.service.model

class Match(private val name: String) {
    var teams: MutableList<Team> = mutableListOf()
    var rule: Rule? = null
    private var id: Int? = null

    constructor(id: Int, name: String) : this(name) {
        this.id = id
    }

    fun getName(): String {
        return this.name
    }

    fun getId(): Int{
        if(this.id == null){
            return 0
        }
        return this.id!!
    }

    fun setId(id: Int): Match{
        this.id = id
        return this
    }
}

File: octopoints_flutter/lib/service/model/Match.dart
import 'package:octopoints_flutter/service/model/Rule.dart';
import 'package:octopoints_flutter/service/model/Team.dart';

class Match {
  final int id;
  final String name;
  List<Team> teams = [];
  List<Rule> rules = [];

  Match({
    this.id = 0,
    required this.name
  });
}
